darts/models/forecasting/regression_model.py

```diff
@@ -1149,6 +1135,44 @@ def _predict_and_sample(
         k = x.shape[0]
         return prediction.reshape(k, self.pred_dim, -1)
 
+    def _extract_component_lags_autoregression(
+        self,
+        cov_type: str,
+        values_matrix: np.ndarray,
+        shift: int,
+        last_step_shift: int,
+        t_pred: int,
+    ) -> np.ndarray:
+        """Extract, concatenate and reorder component-wise lags to obtain a features order
+        identical to tabularization.
+        """
+        # prepare index to reorder features by lags across components
+        comp_lags_reordered = np.concatenate(
+            [
+                np.array(comp_lags, dtype=int)
+                for comp_lags in self.component_lags[cov_type].values()
+            ]
+        ).argsort()
+
+        # convert relative lags to absolute
+        if cov_type == "target":
+            lags_shift = -shift - last_step_shift
+        else:
+            lags_shift = -self.lags[cov_type][0] + t_pred
+
+        # extract features
+        tmp_X = [
+            values_matrix[
+                :,
+                np.array(comp_lags) + lags_shift,
```

Comment:
Nitpicking: for such small arrays it will probably be more efficient with a list comprehension